### PR TITLE
Parallelize decom process across sets

### PR DESCRIPTION
**Pull request by krishnasrinivas**
## Description
1. Disables decom resume during startup (as there were random crashes)
2. decom is done across sets in parallel

Not to be merged with master. This is for hotfix

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

File: cmd/erasure-server-pool-decom.go
Comment (kannappanr) [inline cmd/erasure-server-pool-decom.go]: is the number of sets 14?

Comment (krishnasrinivas) [reply cmd/erasure-server-pool-decom.go]: it's 12 @kannappanr

Comment (harshavardhana) [reply cmd/erasure-server-pool-decom.go]: Just use the len(pool.sets)

Comment (harshavardhana) [reply cmd/erasure-server-pool-decom.go]: This being a configurable value doesn't make sense even as a hotfix.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/erasure-server-pool-decom.go
Comment (poornas) [reply cmd/erasure-server-pool-decom.go]: having a configurable value will allow reducing concurrency if needed below len(pool.sets)

**Review by poornas — APPROVED** [review]
LGTM

File: cmd/erasure-server-pool-decom.go
Comment (harshavardhana) [reply cmd/erasure-server-pool-decom.go]: Then it should default to len(pool.sets) and then allow reducing. Not use 12 as hardcoded value. 

If this is also something we do not merge to master why send this as a PR, just build locally and give it to customer.

File: cmd/erasure-server-pool-decom.go
Comment (kannappanr) [inline cmd/erasure-server-pool-decom.go:724]: should we log something that says that this error happened in decommissioning?

Comment (harshavardhana) [reply cmd/erasure-server-pool-decom.go:724]: That would be automatic since the stack here would indicate that this is inside decom.

File: cmd/erasure-server-pool-decom.go
Comment (kannappanr) [reply cmd/erasure-server-pool-decom.go]: we need to improve upon this a bit more before we can merge it to master. We can default to `len(pool.sets)`

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/erasure-server-pool-decom.go
Comment (kannappanr) [reply cmd/erasure-server-pool-decom.go:724]: displaying in the error string itself, makes it easier to identify

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
Resuming fix sent #14705

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/erasure-server-pool-decom.go
Comment (harshavardhana) [inline cmd/erasure-server-pool-decom.go]: Please rebase don't need to remove this code anymore.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by kannappanr — APPROVED** [review]
LGTM

File: cmd/erasure-server-pool-decom.go
Comment (kannappanr) [reply cmd/erasure-server-pool-decom.go]: done

**Pull request merged**
